etc/kayobe/ansible/cephadm-enter-maintenance.yml

```diff
@@ -0,0 +1,32 @@
+---
+- name: Ensure a Ceph host has entered host maintenance
+  gather_facts: true
+  hosts: ceph
+  become: true
+  tasks:
+    - name: Check if host can enter maintenance mode
+      ansible.builtin.import_role:
+        name: stackhpc.cephadm.commands
+      vars:
+        cephadm_commands:
+          - "orch host ok-to-stop {{ ansible_facts.nodename }}"
+
+    - block:
+        - name: Extract full name of active Ceph manager
+          ansible.builtin.set_fact:
+          active_ceph_mgr: "{{ cephadm_commands_result.results[0].stderr | split | last | replace(\"'\",'') }}"
+
+        - name: Ensure active manager has been switched to another node
+          ansible.builtin.import_role:
+            name: stackhpc.cephadm.commands
+          vars:
+            cephadm_commands:
+              - "mgr fail {{ active_ceph_mgr }}"
+      when: '"Cannot stop active Mgr daemon" in cephadm_commands_result.results[0].stderr'
+
+    - name: Ensure host is in maintenance mode
```

Comment:
I'm not entirely sure how the cpehadm_commands role works, but it might be worth having one last play to ensure the new state is set, and erroring if not